File: Prajwal Dumbre/Tech_round_1/BinaryTreeSize.java
class Node {
    int data;
    Node left, right;

    Node(int data) {
        this.data = data;
        left = right = null;
    }
}

class BinaryTree {
    Node root;

    int getSize(Node node) {
        if (node == null)
            return 0;
        return 1 + getSize(node.left) + getSize(node.right);
    }

    public static void main(String[] args) {
        BinaryTree tree = new BinaryTree();
        tree.root = new Node(10);
        tree.root.left = new Node(20);
        tree.root.right = new Node(30);
        tree.root.left.left = new Node(40);
        tree.root.left.right = new Node(50);

        System.out.println("Size of the tree: " + tree.getSize(tree.root)); // Output: 5
    }
}

File: Saniya_DSA_Codes/binary_tree_size.cpp
#include<iostream>

using namespace std;
struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};


int getsize(Node *root)
    {
        if( root==NULL ) return 0;
        
        int left=getsize(root->left);
        int right=getsize(root->right);
        
        return left+right+1;
        
    }
int getSize(Node* node) {
        // code here
        return getsize(node);
    }


int main()
{
    Node *root=new Node(20);
    Node *one=new Node(10);
    Node *two=new Node(30);
    root->left=one;
    root->right=two;
    int res=getSize(root);
    cout<<"size is:"<<res<<endl;

}